### DISCUSSION TITLE
Welcome to de10-nano Discussions!

### DISCUSSION THREAD

**Discussion by zangman** [zangman/de10-nano#15] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by MorinagaKojyu** [comment]
I want to develop linux using DE10 Nano, but it doesn't work.
I can build the U-Boot and the kernel with the information on this site, but I am unable to build the root file system.
I am trying to follow the instructions, but it is not working.
I get an error during the build process.
One more point, when I boot the debian image I downloaded with DE10-NANO, the USB-OTG
Please tell me how to use ethernet with the USB-OTG function. Thank you in advance.

**Reply by kirkster96** [reply]
Hey @MorinagaKojyu,

I am not sure if you decided to go with the Debian development environment shown in the [guide](https://github.com/zangman/de10-nano/blob/master/docs/Setting-up-the-Development-Environment.md#setting-up-the-development-environment) or if you have something different, but I have created a live video walkthrough of me completing the [Building Embedded Linux - Full Custom](https://github.com/zangman/de10-nano/blob/master/docs/Setting-up-the-Development-Environment.md#setting-up-the-development-environment) guide.

In my video I used Windows 10 and WSL for my environment to complete the steps. Feel free to check it out if you think it will help: https://www.youtube.com/watch?v=zTK2ZQdXhPs

**Comment by kirkster96** [comment]
> To get started, comment below with an introduction of yourself and tell us about what you do with this community.

Hello everyone,

My name is Cameron Kirk and I graduated from UNLV May 2021 with a B.S. in Computer Engineering. I currently work as a software developer in Las Vegas writing primarily IBM assembler code (sometimes C/C++ code) for a mainframe software company. Prior to my current job, I worked part-time as an instructor at Code Central, which is a technology learning center for kids and highschoolers to learn how to write code.

I did mostly FPGA and microcontroller work during my studies and I really enjoy it. I would also consider C++ and Python to be my best programming languages. I make tutorial videos on my [Youtube Channel](https://www.youtube.com/channel/UC_a2S9QtxZ2_IsjrG_-Ll0Q). This GitHub Repo has exposed me to a lot of great knowledge with it's great use of reference links. I will try my best to help out and give back on this Repo where I can! 😊

**Reply by zangman** [reply]
Hi Cameron - Thanks for your contributions with the navs and cleaning the repo to make it easier for everyone to contribute. Great YT channel, I submitted a comment on your video covering this repo, but it didn't show up for some reason :).

**Reply by Andy2No** [reply]
@zangman YouTube comment censorship is now completely arbitrary, and has apparently been handed over to badly written software implementing a constantly changing set of rules that are never disclosed to anyone outside of Google.

They always show the comment to the person who posted it, immediately afterwards, but if you look at the page again while logged out, or in a private window, you'll often see it's not actually shown.  Useful strategies for getting around it include only making very short comments, especially once you see it happen, and deliberately misspelling any word you suspect triggers it.  You may find you can make several really short comments and those get shown, but a longer one containing the full message doesn't.

I'm pretty sure they've decided github is a dirty word, for example, based on my own experience.

**Reply by kirkster96** [reply]
Thank you for checking out my channel! I am confused about what happened with the comment. You are not the first one to tell me this has happened to them on my channel.

When I logged in, I had no new comments on my dashboard. I noticed I had the YouTube default "Hold potentially inappropriate comments for review", but I have never had a comment show up for review. I just changed it to "Allow all comments", but it seems like YouTube is just quietly deleting according to it's own system. Sorry about that! It is out of my control as @Andy2No explains.

However, I am going to continue to encourge my viewers (there is not many) to come over to this Repo to learn more!

**Comment by andkostov** [comment]
Hi all!
I've recently run into an issue when building the second stage of my debootstrap as per the [guide](https://github.com/zangman/de10-nano/blob/master/docs/Debian-Root-File-System.md).

![image](https://user-images.githubusercontent.com/58448051/197378285-343cd74a-711e-4bb1-b46c-436875abf30d.png)

For context here's how I got there:

1. Chose my working environment as Ubuntu WSL (sorry for not going with Debian, just felt like going against the grain :P)
2. Worked out of C:\\Users\\\<username\>\\Document\\de10_nano - this did me no favours as when I ran the first stage of debootstrap it would warn me of my mounting
![image](https://user-images.githubusercontent.com/58448051/197378361-b24fbf1b-f414-433c-9664-19f34d34bc30.png)
3. Move everything to /home/<username> for the "standard" Linux fs - able to run first stage.
4. Subsequently able to `sudo cp /usr/bin/qemu-arm-static rootfs/usr/bin/`, then `sudo chroot rootfs /usr/bin/qemu-arm-static /bin/bash -i`
5. Broke once I ran `/debootstrap/debootstrap --second-stage`

Was just wondering if I could grab some insight on how I might've broken my working environment?

Thanks and kind regards!
Andrei

**Reply by andkostov** [reply]
**FIXED**
This may be obvious to some, but what I needed to do was run `update-binfmts --enable qemu-arm`, meaning you'll need to install the `binfmt-support` package.

Why? Because my PC is x64.

Was able to run the second-stage debootstrap!

**Comment by JasonSaf** [comment]
First of all, many thanks to the contributors of this project. I've found this to be extremely helpful in getting started with the DE10-Nano. I've run into an issue booting Linux on my device that I can't seem to figure out. I've attached a test file from a serial terminal to show what's going on. I get the same "No filesystem could mount root" message for both my own custom Linux copy and the one available for download on this repo. Any suggestions?
[Error.txt](https://github.com/zangman/de10-nano/files/10825933/Error.txt)

**Reply by JasonSaf** [reply]
@zangman @kirkster96

**Reply by zangman** [reply]
Hi it sounds like there might be something wrong with your SD card or the software you are  using to burn to your SD card.

Can you try a different SD card and maybe use Rufus (windows) or dd (linux)?

**Reply by JasonSaf** [reply]
Thank you for taking the time to respond. I appreciate it. After trying both rufus and dd the issue is persisting. I rebuilt the sdcard image as well and it provides the same result. Rufus sets up the default image from Terasic's website just fine so I think those are working okay. I'll try starting from scratch and see if I missed a step.

**Reply by zangman** [reply]
Can you try Rufus with the images from the Releases page and see if that works?

https://github.com/zangman/de10-nano/releases

**Reply by JasonSaf** [reply]
That one works perfectly. I’m able to boot and login just fine.

**Reply by zangman** [reply]
Oh I see - in that case, you might want to check:

 1. Are the partitions set up in the correct order?
 2. Did you set up the root partition correctly?
 3. Have you specified everything correctly in extlinux.conf?

That's where I would start. From your error message, it seems like the Bootloader is working fine and so is the kernel partition. It fails when it tries to read the root partition.

**Comment by lmz0528** [comment]
Hi, I followed the instructions on this page and everything worked fine, thanks a lot.
I found that the storage space is limited, 77% of it was used after I installed it and turned it on.

![圖片1](https://user-images.githubusercontent.com/75722073/223961702-bcca8bd2-5cbb-4e82-b236-3a9cbf8a7f91.png)

Is it possible to use the remaining space of the SD card?
Any help would be much appreciated!

**Reply by lmz0528** [reply]
I found out that setting bs higher when creating the SD card image solved my problem, stupid me!

**Reply by zangman** [reply]
Good stuff, glad you got it working! :)

**Comment by 931759793** [comment]
Hello, I'm a beginner. In the final step of building the kernel, I prompted the following error. How can I fix it ?
![image](https://user-images.githubusercontent.com/24364285/228185763-80a572a7-a23a-4783-a299-f34459318e34.png)

**Reply by zangman** [reply]
Not sure - haven't seen this before. I think you might want to try the steps again.

**Reply by 931759793** [reply]
I reinstalled the Ubuntu system and cross-compiler, and the problem is solved. Thanks for your help.

**Comment by lmz0528** [comment]
Hi, I was wondering if it's possible to get the user interface of Debian with HDMI output and control it with a keyboard and mouse, similar to using the Ubuntu desktop version. I know that Terasic offers an LXDE image file with similar functionality, so I'm curious if Debian can do the same thing. If it's possible, it would essentially give me a Raspberry Pi with FPGA capabilities (if I understand correctly).Any help would be greatly appreciated!

**Reply by zangman** [reply]
It's definitely possible, I just haven't looked into it myself. I believe you will need to enable X11 and have it write to a framebuffer in memory which will then be read by the FPGA and pushed to the HDMI port. Maybe you can copy the RBF file from the LXDE image to use on your installation and see how you can use X11 framebuffer?

PS: I'm just guessing here honestly. I don't know if it's the right approach.

**Comment by MTahaK** [comment]
Hi! Thanks for making this tutorial - it's been very helpful so far. After writing to the SDRAM, I intend to run the data sent over through some modules implemented on the FPGA before writing it back to the SDRAM to be read by the HPS. From what I understand, this would follow a similar approach, but I'm not too sure on the details. Specifically, I'm not sure how I should wire the f2h_axi_slave after enabling the FPGA-HPS AXI Bridge in the HPS module. Do you have any ideas about resources you could point me to? Thanks.

**Comment by NehLamS** [comment]
Thanks for this great work. I try to adapt the flow design GSRD and GHRD for DE10-Nano  (https://www.rocketboards.org/foswiki/Documentation/GSRD1402), and i can understand many things from your tutorial here. Until now i am wolking your tutorial without problem until the partprobe /dev/loop0 i get an error. I use ubuntu under WSL2.

 
![image_error](https://github.com/zangman/de10-nano/assets/92782199/2f7ce1e9-8100-4054-846b-8cd54a9a01d8)

Any idea, and Thanks again for your contribution.

**Reply by RahafQahtani** [reply]
how did you solve it I keep getting the same error

**Comment by Fr0stByte00** [comment]
Anyone know what youtube video he put in the references of the Primer on Avalon page:
[YouTube tutorial - Custom component development using Avalon and AXI](https://www.youtube.com/watch?v=Vw2_1pqa2h0) - Highly recommend watching this video. It is quite long, but explains everything about the interface.
- The video got taken down. If anyone has any other ones they recommend please let me know! TY

**Reply by zangman** [reply]
Ah that's unfortunate, I don't exactly remember what the video contained, but this seems close. You can skim through the section on memory mapped: https://youtu.be/auxFLON7mJo?si=9UtHRXTiUYmnqYWw
